### DISCUSSION TITLE
Keyword with dot(s)

### DISCUSSION THREAD

**Discussion by Arkelis** [hylang/hy#2106] [Q&A] [answered]
Hi, I'm currently trying to mimic [Hiccup](https://github.com/weavejester/hiccup) in Hy. It compiles expressions to HTML (for example `[:div {:class "my-class} "some text"]` gives `<div class="my-class">some text</div>`.

It provides a shortcut for providing classes and ids directly in the tag name, i.e. the previous example can be written as `[:div.my-class "some text"]`. The problem is that in Hy keywords cannot have dots:

```
=> :div.my-class
Traceback (most recent call last):
  File "stdin-ccce5d7095df43c19fa573902762419d4afd243b", line 1
    :div.my-class
    ^
hy.lex.exceptions.LexException: Cannot access attribute on anything other than a name (in order to get attributes of expressions, use `(. <expression> <attr>)` or `(.<attr> <expression>)`)
```

I'm understanding that it conflicts with the attribute access mechanism... What could be a solution to this?

Cheers

**Comment by allison-casey** [comment, marked as answer]
this is currently a limitation of the parser. @scauligi is currently doing a full rewrite of the parser that would give us proper support for compile time reader macros and could also potentially give us the flexibility to fix this as well and its already an open issue (see #2096). until then though, you will likely not be able to implement the keyword sugar for class/ids that hiccup provides unless you want to also allow/switch to strings as the tag name (i.e. `["div.my-class#my-id" ...]` which is what i'd probably do) or stick with the explicit dict passing. 

As an aside, using keywords as dictionary keys in Hy is generally discouraged since Hy is much more closely related to python than clojure is to java and since keywords are used for kwargs such that this is true 
```clojure
(dict :class "my-class")  ; => {"class" "my-class"}
(:class (dict :class "my-class"))  ; => "my-class"
````

**Reply by Arkelis** [reply]
Thanks for your answer! You're right, as the keyword is not a "Python concept", I should rely on strings.
